usr/share/rear/rescue/default/500_ssh.sh

```diff
@@ -54,5 +57,110 @@ if has_binary sshd; then
         *     ) echo "root:$(echo $SSH_ROOT_PASSWORD | openssl passwd -1 -stdin):::::::" > $ROOTFS_DIR/etc/shadow ;;
         esac
     fi
-fi
 
+    # Part 2: SSH client (ssh) - this is for accessing other systems (e.g. the backup server) via SSH from the recovery system
+
+    Log "Copying SSH client configuration"
+
+    local target_dir="$ROOTFS_DIR/root/.ssh" original_file target_file passphrase_set="no" passphrase="" repeated_passphrase
+
+    mkdir -p -m u=rwx,go=- "$target_dir"
+
+    for original_file in /root/.ssh/id_* /root/.ssh/known_host[s]; do
+        case "$original_file" in
+            (*/id_*.pub|*/known_hosts)
+                # Public key files or 'knownhosts' contain no secrets and can be copied as-is
+                Log "Copying $original_file"
+                cp -p "$original_file" "$target_dir"
+                ;;
+
+            (*/id_*)
+                # Private key files will be copied, examined and, if desired, passphrase-protected
+
+                target_file="$target_dir/$(basename "$original_file")"
+                cp -p "$original_file" "$target_file"
+
+                # This ssh-keygen invocation checks if a passphrase change succeeds with an empty passphrase for
+                # authentication. If so, the private key file is not currently passphrase-protected.
+                if ssh-keygen -q -p -P '' -N '' -f "$target_file" >/dev/null 2>&1; then
+                    # Handle the unprotected private key file
+
+                    # Obtain a decision on copying and passphrase-protection once
+                    if [ "$passphrase_set" == "no" ]; then
+                        passphrase_set="yes"
+
+                        LogUserOutput ""
+                        LogUserOutput "WARNING: User 'root' on '$HOSTNAME' has potential SSH access to other systems via"
+                        LogUserOutput "unprotected private keys. To copy those keys onto the rescue medium in protected form,"
+                        LogUserOutput "please enter a strong passphrase which will be used when running 'rear recover'."
+                        LogUserOutput "- Entering an empty passphrase will *not* copy unprotected private keys to the rescue medium."
+                        LogUserOutput "- Entering 'UnProtected' (without quotes) will copy unprotected private keys onto the rescue medium."
+                        LogUserOutput ""
+
+                        while true; do
+                            LogUserOutput "SSH client passphrase (or empty, or 'UnProtected'): \c"
+                            read -s passphrase <&6
```

Comment:
Can't you use the new `UserInput` function for this? That way you could also handle setting an answer for it.

Reply:
I did first, but then `UserInput` would copy the passphrase into the log file as plain text, which is a no-go. So doing so would require a careful change to `UserInput`.

Reply:
Yes, go ahead and improve `UserInput` if you need it.

Reply:
I will enhance the UserInput function to be silent
if requested (i.e. via an additional option).